rtl/include/eclhelper.hpp

```diff
@@ -780,8 +780,312 @@ interface ICodeContext : public IResourceContext
     virtual IEngineRowAllocator * getRowAllocatorEx(IOutputMetaData * meta, unsigned activityId, unsigned flags) const = 0;
     virtual void addWuExceptionEx(const char * text, unsigned code, unsigned severity, unsigned audience, const char * source) = 0;
     virtual unsigned getElapsedMs() const = 0;
+    virtual unsigned getWorkflowId() const = 0; // Note: don't use yet as it has not been fully implemented in all derived classes
 };
 
+class IndirectCodeContext : implements ICodeContext
```

Comment:
This needs a export qualifier otherwise you may get link problem in windows, or with release versions.

`class ECLRTL_API IndirectCodeContext : implements ICodeContext
`